spec/attributes.md

```diff
@@ -162,84 +162,67 @@ The types of positional and named parameters for an attribute class are limited
 Attributes are specified in ***attribute sections***. An attribute section consists of a pair of square brackets, which surround a comma-separated list of one or more attributes. The order in which attributes are specified in such a list, and the order in which sections attached to the same program entity are arranged, is not significant. For instance, the attribute specifications `[A][B]`, `[B][A]`, `[A,B]`, and `[B,A]` are equivalent.
 
 ```antlr
-global_attributes
-    : global_attribute_section+
-    ;
-
-global_attribute_section
-    : '[' global_attribute_target_specifier attribute_list ']'
-    | '[' global_attribute_target_specifier attribute_list ',' ']'
-    ;
-
-global_attribute_target_specifier
-    : global_attribute_target ':'
-    ;
-
-global_attribute_target
-    : 'assembly'
-    | 'module'
-    ;
-
-attributes
-    : attribute_section+
-    ;
-
-attribute_section
-    : '[' attribute_target_specifier? attribute_list ']'
-    | '[' attribute_target_specifier? attribute_list ',' ']'
-    ;
-
-attribute_target_specifier
-    : attribute_target ':'
-    ;
-
-attribute_target
-    : 'field'
-    | 'event'
-    | 'method'
-    | 'param'
-    | 'property'
-    | 'return'
-    | 'type'
-    ;
-
-attribute_list
-    : attribute (',' attribute)*
-    ;
-
-attribute
-    : attribute_name attribute_arguments?
-    ;
-
-attribute_name
-    : type_name
-    ;
-
-attribute_arguments
-    : '(' positional_argument_list? ')'
-    | '(' positional_argument_list ',' named_argument_list ')'
-    | '(' named_argument_list ')'
-    ;
-
-positional_argument_list
-    : positional_argument (',' positional_argument)*
-    ;
-
-positional_argument
-    : attribute_argument_expression
-    ;
-
-named_argument_list
-    : named_argument (','  named_argument)*
-    ;
-
-named_argument
-    : identifier '=' attribute_argument_expression
-    ;
-
-attribute_argument_expression
-    : expression
-    ;
+global_attributes: 
+    global_attribute_section+;
+
+global_attribute_section:
+     '[' global_attribute_target_specifier attribute_list ']' |
+     '[' global_attribute_target_specifier attribute_list ',' ']';
```

Comment:
why was this changed?  The previous way is much easier to read.

Reply:
specifically, having each production start with either `:` or `|` scans really well.  It's also how we generate our grammar files here: https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Generated/CSharp.Generated.g4